ReactAndroid/build.gradle

```diff
@@ -5,10 +5,15 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+buildscript {
+    dependencies {
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:${rootProject.hasProperty("kotlinVersion") ? rootProject.ext.kotlinVersion : KOTLIN_VERSION}"
+    }
+}
```

Comment:
I think this will have unwanted side effects. Specifically mixing `buildscript{}` and `plugins{}` in Gradle is not advised. 

It seems like a workaround that we could live with 🤔 But I would look for a solution that uses only `plugins{}` if possible.

Reply:
As far as I looked, having dynamic version in plugins isn't quite possible. Since, we require to use the kotlinVersion from root level if that's present.

See [here](https://docs.gradle.org/current/userguide/plugins.html#sec:constrained_syntax)

Reply:
Can you add a note on why this is needed otherwise we'll forget about it

Reply:
So the reason to go with this approach is: 

use the **kotlinVersion** defined in top level build.gradle. If it's not defined, we use the default(1.6.10 as of now) defined in gradle.properties. We couldn't add dynamic version using the plugins DSL because of the constrained syntax, so we had to introduce the buildScript for this. See [here](https://docs.gradle.org/current/userguide/plugins.html#sec:constrained_syntax)

Should I add it as a comment in the build.gradle or are we good here?

Reply:
We're good here 👍 
I'm looking into cleaning this up a bit in the future so this will hopefully go.